services/submission/src/main/java/app/coronawarn/server/services/submission/config/SubmissionServiceConfig.java

```diff
@@ -36,6 +38,7 @@
 
   private static final String PATH_REGEX = "^[/]?[a-zA-Z0-9_]+[/[a-zA-Z0-9_]+]*$";
   private static final String URL_WITH_PORT_REGEX = "^http[s]?://[a-z0-9-]+(\\.[a-z0-9-]+)*(:[0-9]{2,6})?$";
+  private static final String ALLOWED_COUNTRY_CODES_REGEX = "^([a-zA-Z]{2}(\\,*[a-zA-Z]{2})*)$";
```

Comment:
Looking at `[a-zA-Z]{2}`:

When you allow lower and uppercase, all country code compare functions need to be case insensitive.

When the country code compare function need to be case insensitive, then there need to be test cases that ensure/test case insensitive.

Off-course: only allowing either upper or lower case also requires test case - but at least to me input sanitizing sounds easier then tracing down issues with different environments, inputs, code execution paths, string compares (both Java and SQL) etc.

Reply:
You make a good point about the potential issues that could appear with this kind of relaxation of constraints around the input.
I've also given thought to it during implementation, but because the ISO specification doesn't enforce lower/upper cases for these country codes, i decided to make this startup input a little more robust for any scripts or infrastructure that provides them (plan is to expose as app parameter not just statically defined in the yaml)
The check/logic done against this variable is case ignorant (see SubmissionServiceConfig.isCountryCodeAllowed), however, you're right, and i do plan to  add more test cases to cover this.
I will analyze again the potential risks though.